File: server/src/models/ParkingSpot.js
// Модель парковочного места — статус, зона, привязка к устройству
const mongoose = require('mongoose');

const parkingSpotSchema = new mongoose.Schema(
  {
    parkingLotId:    { type: mongoose.Schema.Types.ObjectId, ref: 'ParkingLot', required: true },
    spotNumber:      { type: Number, required: true },
    status:          { type: String, enum: ['free', 'occupied', 'reserved', 'maintenance'], default: 'free' },
    deviceId:        { type: String, default: null },  // ID сенсора ESP32
    floor:           { type: Number, default: 1 },
    zone:            { type: String, default: 'A' },
    isSubscription:  { type: Boolean, default: false }, // закреплено за абонементом
    subscribedUserId:{ type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
  },
  { timestamps: true }
);

// Уникальный номер места внутри парковки
parkingSpotSchema.index({ parkingLotId: 1, spotNumber: 1 }, { unique: true });

module.exports = mongoose.model('ParkingSpot', parkingSpotSchema);

File: mobile/lib/models/parking_spot.dart
// Модель парковочного места
class ParkingSpot {
  final String id;
  final String parkingLotId;
  final int spotNumber;
  final String status; // free, occupied, reserved, maintenance
  final String? deviceId;
  final int floor;
  final String zone;
  final bool isSubscription;
  final String? subscribedUserId;

  ParkingSpot({
    required this.id,
    required this.parkingLotId,
    required this.spotNumber,
    this.status = 'free',
    this.deviceId,
    this.floor = 1,
    this.zone = 'A',
    this.isSubscription = false,
    this.subscribedUserId,
  });

  factory ParkingSpot.fromJson(Map<String, dynamic> json) => ParkingSpot(
        id: json['_id'] ?? json['id'] ?? '',
        parkingLotId: json['parkingLotId'] is Map
            ? json['parkingLotId']['_id'] ?? ''
            : json['parkingLotId'] ?? '',
        spotNumber: json['spotNumber'] ?? 0,
        status: json['status'] ?? 'free',
        deviceId: json['deviceId'],
        floor: json['floor'] ?? 1,
        zone: json['zone'] ?? 'A',
        isSubscription: json['isSubscription'] ?? false,
        subscribedUserId: json['subscribedUserId'],
      );

  bool get isFree => status == 'free';
  bool get isOccupied => status == 'occupied';
  bool get isReserved => status == 'reserved';
  bool get isMaintenance => status == 'maintenance';
}
